plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/actions/ObjectPropertyTester.java

```diff
@@ -129,12 +132,15 @@ public boolean test(Object receiver, String property, Object[] args, Object expe
             }
             case PROP_CAN_DELETE: {
                 if (node instanceof DBNDataSource || node instanceof DBNLocalFolder) {
-                    return node.getOwnerProject().hasRealmPermission(RMConstants.PERMISSION_PROJECT_CONNECTIONS_EDIT);
+                    return nodeProjectHasPermission(node, RMConstants.PERMISSION_PROJECT_CONNECTIONS_EDIT);
                 }
 
                 if (DBNUtils.isReadOnly(node)) {
                     return false;
                 }
+                if (node instanceof DBNNodeWithResource && !nodeProjectHasPermission(node, RMConstants.PERMISSION_PROJECT_RESOURCE_EDIT)) {
```

Comment:
Perhaps the `nodeProjectHasPermission` method should care about `node` being an instance of `DBNNodeWithResource`?

Reply:
No, this method is used not only for DBNNodeWithResource.